src/Razor/Microsoft.AspNetCore.Razor.Language/src/Extensions/DefaultTagHelperTargetExtension.cs

```diff
@@ -650,7 +650,7 @@ internal static string GetDeterministicId(CodeRenderingContext context)
 
         private static string GetPropertyAccessor(DefaultTagHelperPropertyIntermediateNode node)
         {
-            var propertyAccessor = $"{node.FieldName}.{node.PropertyName}";
+            var propertyAccessor = string.Concat(node.FieldName, '.', node.PropertyName);
```

Comment:
Concat -> +

The compiler will change + to the most efficient use of Concat for you.

Reply:
This might avoid boxing, unlike the `char` constant:
```suggestion
            var propertyAccessor = string.Concat(node.FieldName, ".", node.PropertyName);
```

Reply:
It's interesting that C# doesn't do this for us --> [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAmARgFgAoQgZgAIS6BhOgbyrs4fsNIAY6AWVIAKXgICGaOmAAWEqHWDTxMgJQcuWgLwA+BvwB0TCADswEjCKlLpYNQG5NnZ9wMDBxMfzo3Vy93VXHX1xYzMLKxsA+ydKLVdaQMEab0k7eUUA1Xtgrj1fOgBqJWKZOIBfIA)

I believe this is tracked by https://github.com/dotnet/roslyn/issues/44168 et.al.